e2e-playwright/start-server

```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+# Ensure we're in the Grafana root directory
+cd "$(dirname "$0")/.."
```

Comment:
is there a better way to do this? 🤔

Reply:
maybe `cd "$(dirname "${BASH_SOURCE[0]}")/.." || exit 1`

${BASH_SOURCE[0]} is a little more reliable because:
- $0 can be misleading if the script is sourced (e.g., . start-server) - it shows the shell name instead of the script path
- ${BASH_SOURCE[0]} always refers to the actual script file, regardless of how it's invoked

Also adding || exit 1 ensures the script fails fast if the directory doesn't exist, rather than continuing in the wrong location

These changes makes it more robust for different execution contexts (direct execution vs sourcing) and better error handling.